apps/judicial-system/backend/src/app/modules/case/case.controller.ts

```diff
@@ -460,6 +460,33 @@ export class CaseController {
     return theCase
   }
 
+  @UseGuards(JwtAuthGuard, CaseExistsGuard)
+  @RolesRules(
+    districtCourtJudgeRule,
+    districtCourtRegistrarRule,
+    districtCourtAssistantRule,
+  )
+  @Get('case/:caseId/connectedCases')
+  @ApiOkResponse({ type: [Case], description: 'Gets all connected cases' })
+  async getConnectedCases(
+    @Param('caseId') caseId: string,
+    @CurrentCase() theCase: Case,
+  ): Promise<Case[]> {
+    this.logger.debug(`Getting connected cases for case ${caseId}`)
+
+    if (!theCase.defendants || theCase.defendants.length === 0) {
+      return []
+    }
+
+    const connectedCases = await Promise.all(
+      theCase.defendants.map((defendant) =>
```

Comment:
Wasn't the idea to only support cases with a single defendant?

Reply:
Yes you're right. My thinking here was to make this more generic than that specific use case as it felt a bit obscure to me to have a function called "getConnectedCases" that throws an error if there's more than one defendant, rather than returning all relevant information and handling the filtering of it higher up where we know what we want out of it. Maybe I'm overthinking it though for future use cases we aren't handling yet 😅 What do you think?

Reply:
I think you are right.